include/shared.h

```diff
@@ -277,11 +277,11 @@ size_t datatype_to_size(enum fi_datatype datatype);
 #endif
 
 #define FT_EQ_ERR(eq, entry, buf, len) \
-	FT_ERR("eq_readerr: %s", fi_eq_strerror(eq, entry.prov_errno, \
+	FT_ERR("eq_readerr: %s", fi_eq_strerror(eq, entry.err, \
 				entry.err_data, buf, len))
 
 #define FT_CQ_ERR(cq, entry, buf, len) \
-	FT_ERR("cq_readerr: %s", fi_cq_strerror(cq, entry.prov_errno, \
+	FT_ERR("cq_readerr: %s", fi_cq_strerror(cq, entry.err, \
```

Comment:
Let's print both errors.

Reply:
Note that the prov_errno needs to be printed as an integer, since we can't interpret it.